### PR TITLE
Added Factory implementation to create drive client

### PR DESCRIPTION
**Pull request by stiartsly**

### CODE REVIEW

File: src/client.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    HiveDriveType_ownCloud  = 0x51,
+
+    HiveDriveType_HiveIPFS  = 0x98,
+    HiveDriveType_Butt      = 0x99
+};
+
+
+typedef struct HiveOptions {
+    char perisisten_location;
```
Comment (lifenxiang) [inline src/client.h]: typo

Comment (lifenxiang) [inline src/client.h]: typo

File: src/client.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    int  drive_type;
+} HiveOptions;
+
+typedef struct OneDriveOptions {
+    HiveOptions base;
+
+    const char *client_id;
```
Comment (lifenxiang) [inline src/client.h]: these three fields should be handled by app

File: src/client.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    const char *scope;
+    const char *redirect_url;
+
+    int (*grant_authorize)(const char *request_url);
+} OneDriveOptions;
+
+HiveClient *hive_client_new(const HiveOptions *options);
+
```
Comment (lifenxiang) [inline src/client.h]: add an api for user login?

Comment (stiartsly) [reply src/client.h]: yes, will be

File: src/client.h (hunk 2 above)
Comment (stiartsly) [reply src/client.h]: Yes, these all fields should be filled by apps.

**Pull request merged**